### DISCUSSION TITLE
09.01.2024

### DISCUSSION THREAD

**Discussion by Roadraudi** [Roadraudi/Codespace_Repository#10] [Announcements]
Happy new year!

Bezüglich der Woche vor Weihnachten habe ich noch einen [Rechner](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Benzinverbrauch.py) programmiert, der mir meinen Benzinverbrauch zurück liefert sowie ein Programm für die  [Fibonacci-Zahl](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Fibonacci_zahl.py)die die Anzahl an Stellen zurück gibt, die ich als Eingabe tätige. 

In der gestrigen Vorlesung haben wir uns intensiver mit den If-Abfragen und den jeweiligen Conditions beschäftigt woraufhin sich auch der folgende Code, ein [Schere-Stein-Papier](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Rock-Paper-Scissors.py) Spiel, orientiert hat. Dieser Quellcode ist aber schon vor der Vorlesung entstanden und war bereits eine Idee aus meinem letzten Announcement, weswegen ich ihn auch hochgeladen habe. Zu guter Letzt komm noch ein [Binär_Rechner](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/DecimalToBinary.py) der Dezimalzahlen in Binärzahlen umwandelt. Dabei habe mit listen gearbeitet, kann ich nur empfehlen da sie nicht wie in Java eine feste Länge haben und sich so sehr vielfältig einsetzen lassen.

Noch als kleine Aussicht habe ich mir überlegt, einen Passwortgenerator zu programmieren, da ich bisher noch auf altmodische Art Passwörter selbst erstellt habe und vermute, dass ich diesen sehr dringend bräuchte.

**Comment by katharinaheller** [comment]
Deine Programmierprojekte klingen spannend und vielfältig. Besonders interessant finde ich den Binär-Rechner mit Listen und die Idee für einen Passwortgenerator. Moderne Passwortgeneratoren sind sicherer, da sie komplexe und zufällige Zeichenkombinationen erstellen, die schwer zu erraten sind. Dies erschwert Angriffe durch Brute-Force-Methoden erheblich, da die Vielzahl möglicher Kombinationen die Erfolgsaussichten deutlich verringert. 
Deshalb, weiter so und viel Erfolg bei deinen zukünftigen Projekten! :)
Grüße

**Comment by okh-mzny** [comment]
Ein gut geschriebener Blogpost und sauberer Code 👍 
Viel Glück bei dem Passwortgenerator und ich bin definitiv gespannt darauf, diesen zu sehen sehen, sobald du ihn fertiggestellt hast. :)

Viele Grüße
